### PR TITLE
[JB][OPS-12609] email address page

### PR DESCRIPTION
**Pull request by barnesjake**

### CODE REVIEW

File: app/uk/gov/hmrc/cardpaymentfrontend/forms/EmailAddressForm.scala [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+object EmailAddressForm {
+
+  val emailAddressKey: String = "email-address"
+
+  private val emailAddressRegex: Regex = """^[a-zA-Z0-9\.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?(?:\.[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?)*$""".r
+
+  def form(): Form[EmailAddress] = {
+    val emailAddressMapping = Forms.of(new Formatter[EmailAddress]() {
+
```
Comment (barnesjake) [inline app/uk/gov/hmrc/cardpaymentfrontend/forms/EmailAddressForm.scala:34]: This approach differs slightly from how it was done in pay-frontend. I chose this way in favour of keeping it within the form method and I _think_ it's a bit simpler, i.e. just a few if else's instead of using constraints etc. Happy to tweak this if the consensus is the constraint implementation is preferred.

TL;DR: Slightly different to pay-frontend implementation, but functionally the same

File: app/uk/gov/hmrc/cardpaymentfrontend/views/Views.scala [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.cardpaymentfrontend.views
+
+import uk.gov.hmrc.cardpaymentfrontend.views.html.EmailAddressPage
+
+import javax.inject.Inject
+
```
Comment (barnesjake) [inline app/uk/gov/hmrc/cardpaymentfrontend/views/Views.scala:22]: This is something I've used/seen in other projects as a simple(er?) way to access the views from a common place, instead of injecting every view we need to each controller, instead you just use this to access the views.

File: app/uk/gov/hmrc/cardpaymentfrontend/forms/EmailAddressForm.scala (hunk 1 above)
Comment (adwelly) [reply app/uk/gov/hmrc/cardpaymentfrontend/forms/EmailAddressForm.scala:34]: Will this form ever b e referenced outside of the Controller? Should we consider keeping it _with_ the Controller?

Comment (barnesjake) [reply app/uk/gov/hmrc/cardpaymentfrontend/forms/EmailAddressForm.scala:34]: Probably not, but my opinion is having it in it's own object keeps the controller code cleaner/simpler. I suppose we could put it into a companion object for the controller. I'm not fussed either way.

File: app/uk/gov/hmrc/cardpaymentfrontend/views/Views.scala (hunk 2 above)
Comment (adwelly) [reply app/uk/gov/hmrc/cardpaymentfrontend/views/Views.scala:22]: I'm doubtful about this. In pay-frontend there's a collection of forms all held together despite the fact that they have nothing in common beyond being Forms. Someone coming to this as a new codebase will have to go through the extra step of accessing the extra Views construct instead of being able to see exactly which views are used here by looking at what is injected. Form that POV this is more of a cognitive load for newcomers and I'm not clear what the advantage is.

File: test/uk/gov/hmrc/cardpaymentfrontend/forms/EmailAddressFormSpec.scala [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+      result.hasErrors shouldBe false
+      result.errors shouldBe List.empty[String]
+    }
+    "should throw error when an invalid email is entered" in {
+      val result: Form[EmailAddress] = form.bind(Map("email-address" -> "not_a_valid_email"))
+      result.hasErrors shouldBe true
+      result.errors shouldBe List(FormError("email-address", List("email-address.error.invalid")))
+    }
+  }
+}
```
Comment (adwelly) [inline test/uk/gov/hmrc/cardpaymentfrontend/forms/EmailAddressFormSpec.scala:54]: Could you add some tests about the regex please.

Comment (barnesjake) [reply test/uk/gov/hmrc/cardpaymentfrontend/forms/EmailAddressFormSpec.scala:90]: Added

File: app/uk/gov/hmrc/cardpaymentfrontend/views/Views.scala (hunk 2 above)
Comment (barnesjake) [reply app/uk/gov/hmrc/cardpaymentfrontend/views/Views.scala:22]: I'll remove this then

File: conf/messages [hunk 4]
```diff
@@ -1,2 +1,91 @@
 service.name = card-payment-frontend
 service.text = This is your new service
+
+
+## Non page specific
+continue = Continue
+error.there-is-a-problem = There is a problem
+
+
+## page titles
```
Comment (barnesjake) [inline conf/messages:10]: Sorry started adding these and forgot to stash it before committing, I can remove if needs be

**Review by michaelgroves90 — APPROVED** [review]
